src/http-adapter/index.js

```diff
@@ -0,0 +1,228 @@
+   * @param {HTTP_METHOD} options.method - the method of the request
+   * @param {Object} options.headers - A key/value object of headers to be sent with the request
+   * @param {Object} options.data - A key/value payload to be sent with the request. If method is GET, these
+   * options will be transformed into query paramters
+   * @param {boolean} options.isInternal - Whether a request is being sent to an internal server
+   * or not to invoke request/response interceptors
+   * @returns {Promise} Promise relating to request.
+   */
+  request (
+    url,
+    options = {}
+  ) {
+    const optionsWithDefaults = Object.assign({
+      method: HTTP_METHOD.GET,
+      headers: {},
+      // set internal to true if this request in intended to go to an internal server and invoke the interceptors
+      internal: false
+    }, options)
+
+    // pass in data, even if none is defined to preverve _http method
+    // pass in headers, even if none are defined to override default headers that are set with the _http method
+    return this._http(
+      optionsWithDefaults.method,
+      url,
+      optionsWithDefaults.data,
+      optionsWithDefaults.headers,
+      optionsWithDefaults.internal
+    )
+  }
+
+  /**
+   * Check if requestion serialization is turned on. If so, the current request will
+   * wait for previous requests to settle. Otherwise, the request will just be executed
+   * @param {Object} options - request options to be propagated to _makeRequest
+   * @param {string} options.url - the url of the request
+   * @param {HTTP_METHOD} options.method - the method of the request
+   * @param {Object} options.headers - A key/value object of headers to be sent with the request
+   * @param {Object} options.data - A key/value payload to be sent with the request. If method is GET, these
+   * options will be transformed into query paramters
+   * @param {boolean} options.isInternal - Whether a request is being sent to an internal server
+   * or not to invoke request/response interceptors
+   * @returns {Promise} Promise relating to request resolution.
+   */
+  _checkSerializeRequests ({
+    url,
+    method,
+    headers,
+    data,
+    isInternal
+  }) {
+    let promise
+    if (this.serializeRequests) {
+      // Wait for all requests to settle (either with success or rejection) before making request
+      const promises = Array.from(this._outstandingRequests).map((promise) =>
+        promise.catch(() => { })
+      )
+
+      promise = Promise.all(promises).then(() =>
+        this._makeRequest({ url, method, headers, data, isInternal })
+      )
+    } else {
+      promise = this._makeRequest({ url, method, headers, data, isInternal })
+    }
+
+    this._outstandingRequests.add(promise)
+    const removeFromOutstandingRequests = () => {
+      this._outstandingRequests.delete(promise)
+    };
+    promise.then(removeFromOutstandingRequests, removeFromOutstandingRequests);
+
+    return promise
```

Comment:
I think it might be super unlikely since the mapped array occurs synchronously. Since this logic is already available in the current HTTPAdapter, I would hope we would have seen issues by now 😅 . I think one of the applications only uses this in one place.